client/src/components/MultiJurisdictionAudit/RoundManagement/useBatchResults.ts

```diff
@@ -0,0 +1,124 @@
+import { useEffect, useState } from 'react'
+import { toast } from 'react-toastify'
+import { api } from '../../utilities'
+
+export interface IResultValues {
+  [batchId: string]: {
+    [choiceId: string]: number | string
+  }
+}
+
+export interface IBatch {
+  id: string
+  name: string
+  numBallots: number
+  auditBoard: null | {
+    id: string
+    name: string
+  }
+}
+
+const stringifyPossibleNull = (v: string | number | null) => (v ? `${v}` : '')
+
+const reformatResults = (r: IResultValues, numberify = true): IResultValues => {
+  return Object.keys(r).reduce(
+    (a, batchId) => ({
+      ...a,
+      [batchId]: Object.keys(r[batchId]).reduce(
+        (b, choiceId) => ({
+          ...b,
+          [choiceId]: numberify
+            ? Number(r[batchId][choiceId])
+            : stringifyPossibleNull(r[batchId][choiceId]),
+        }),
+        {}
+      ),
+    }),
+    {}
+  )
+}
```

Comment:
i've noticed this pattern and i have an idea about a simpler way to do it - if we convert to strings when you plug the data into the actual DOM components, and then convert back to numbers in the onChange event handler, I think it will be a lot more concise. what do you think?

Reply:
@jonahkagan We'd be replacing the native Formik event handler which is conveniently handled magically currently. It's doable, but better to do it inside our form field wrapper across the board, and that's a relatively huge refactor. I'm not averse, just I think it's out of scope for this.

Reply:
ahhh right that makes sense. i have to say that so far I don't really understand what Formik does for us besides automatically running validation and inserting those error messages. the magic state updates don't seem that helpful, and I've often struggled to figure out how to integrate the formik components with the blueprint ones. are there things formik does that I'm just not appreciating?

agree it's not in scope to refactor everything here, though if you wanted to just try it out once, I think even replacing the formik event handler would be more concise haha

Reply:
I'll re-take a look at what all Formik is providing us and refresh my memory as well. :)